File: ahoy-ui/src/app/settings/docker-settings/docker-settings.ts
export class DockerSettings {
  id: number;
  dockerRegistries: DockerRegistry[];

  constructor(id: number) {
    this.id = id;
  }
}

export class DockerRegistry {
  id: number;
  name: string;
  server: string;
  username: string;
  password: string;
  secure: boolean;
}

File: ahoy-server/src/main/java/za/co/lsd/ahoy/server/docker/DockerSettings.java
package za.co.lsd.ahoy.server.docker;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
public class DockerSettings {
	@Id
	private Long id;
	@OneToMany(mappedBy = "dockerSettings", cascade = CascadeType.ALL, orphanRemoval = true)
	@JsonManagedReference
	private List<DockerRegistry> dockerRegistries;
}
